File: Study Plan/Top Interview 150/0058 Length of Last Word/LengthOfLastWord.java
/*  
    Problem
    --------------------------------------------------------------------------------
    58. Length of Last Word
*/

/*  
    Solution
    --------------------------------------------------------------------------------
    https://leetcode.com/problems/find-the-index-of-the-first-occurrence-in-a-string/solutions/4384285/c-time-complexity-o-m-n-space-complexity-o-1-beats-71-in-runtime/
    - Runtime 0 ms (Beats 100.00% of users with Java)
    - Memory 41.23 MB (Beats 75.76% of users with Java)
*/

class Solution {
    public int lengthOfLastWord(String s) {
        String[] words = s.split(" ");
        String last = words[words.length - 1].trim();
        return last.length();
    }
}

File: Study Plan/Top Interview 150/0058 Length of Last Word/LengthOfLastWord.cs
/*  
    Problem
    --------------------------------------------------------------------------------
    58. Length of Last Word
*/

/*  
    Solution
    --------------------------------------------------------------------------------
    - Runtime 42 ms (Beats 84.14% of users with C#)
    - Memory 38.35 MB (Beats 16.20% of users with C#)
*/

public class Solution
{
    public int LengthOfLastWord(string s)
    {
        var words = s.Split(" ", StringSplitOptions.RemoveEmptyEntries);
        var last = words[^1];
        
        return last.Length;
    }
}